roles/cluster_infra/templates/resources.tf.j2

```diff
@@ -54,6 +54,63 @@ resource "openstack_blockstorage_volume_v3" "data_volume" {
   {% endif %}
 }
 
+#####
+##### Cluster networks
+#####
+
+# Get existing network resource data by name
+data "openstack_networking_network_v2" "cluster_network" {
+  name = "{{ cluster_network }}"
+}
+
+# Storage network 
+{% if cluster_storage_network is defined %}
+data "openstack_networking_network_v2" "cluster_storage" {
+  name = "{{ cluster_storage_network }}"
+}
+{% endif %}
+
+#####
+##### Cluster ports
+#####
+
+# Primary network
+resource "openstack_networking_port_v2" "cluster" {
+  name = "{{ cluster_name }}-0"
+  network_id = "${data.openstack_networking_network_v2.cluster_network.id}"
+  admin_state_up = "true"
+
+  security_group_ids = [
+    "${openstack_networking_secgroup_v2.cluster_secgroup.id}",
+  ]
+
+  binding {
+    vnic_type = "{{ cluster_vnic_type | default('normal') }}"
+    {% if cluster_vnic_profile is defined %}
+    profile = <<EOF
+    {{ cluster_vnic_profile | to_json }}
+    EOF
+    {% endif %}
```

Comment:
I don't think we should allow profile here. A correctly configured cloud shouldn't need it, and it is usually an admin-only operation to set it.